File: packages/stripe_android/android/src/main/kotlin/com/facebook/react/uimanager/SimpleViewManager.java
package com.facebook.react.uimanager;

import android.view.View;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.ReadableArray;

import java.util.HashMap;
import java.util.Map;

public abstract class SimpleViewManager<T extends View> {

    public abstract String getName();

    public Map<String, Object> getExportedCustomDirectEventTypeConstants() {
        return new HashMap<>();
    }

    public abstract T createViewInstance(ThemedReactContext reactContext);

    public void onDropViewInstance(@NonNull T view) {};

    public void receiveCommand(T root, String commandId, ReadableArray args) {}

    public void onAfterUpdateTransaction(@NonNull T view) {}
}

File: packages/stripe_android/android/src/main/kotlin/com/facebook/react/uimanager/SimpleViewManager.kt
package com.facebook.react.uimanager

import android.view.View
import com.facebook.react.bridge.ReadableArray

abstract class SimpleViewManager<T: View> {

    abstract fun getName(): String

    open fun getExportedCustomDirectEventTypeConstants(): MutableMap<String, Any> = mutableMapOf()

    abstract fun createViewInstance(reactContext: ThemedReactContext): T

    open fun onDropViewInstance(view: T) {}

    open fun receiveCommand(root: T, commandId: String?, args: ReadableArray?) {}

    open fun onAfterUpdateTransaction(view: T) {}
}
